apps/builder/app/builder/builder.tsx

```diff
@@ -208,6 +223,108 @@ const NavigatorPanel = ({
   );
 };
 
+const revealAnimation = ({
+  show,
+  backgroundColor,
+}: {
+  show: boolean;
+  backgroundColor: string;
+}): CSS => ({
+  position: "relative",
+  "> ::after": {
+    content: "",
+    position: "absolute",
+    inset: 0,
+    zIndex: 1,
+    transitionDuration: "300ms",
+    pointerEvents: "none",
+    transitionProperty: "opacity",
+    backgroundColor,
+    opacity: show ? 0 : 1,
+  },
+});
+
+const useLoadingState = () => {
+  type State =
+    | "dataLoadingState"
+    | "selectedInstanceRenderState"
+    | "canvasIframeState";
+  const [states, setStates] = useState<Map<State, "initial" | "ready">>(
+    new Map([
+      ["dataLoadingState", "initial"],
+      ["selectedInstanceRenderState", "initial"],
+      ["canvasIframeState", "initial"],
+    ])
+  );
+
+  const dataLoadingState = useStore($dataLoadingState);
+  const selectedInstanceRenderState = useStore($selectedInstanceRenderState);
+  const canvasIframeState = useStore($canvasIframeState);
```

Comment:
Why this 3 states are selected as something what have 1/3 of progress. Let's be simple leave `canvasIframeState`

Reply:
IMO this has so low value and so many code, that should be simplified.